### PR TITLE
Final workflow fixes.

### PR DESCRIPTION
**Pull request by unbohn**
This PR includes final workflow fixes to enable the passing of all CLI tests in the `dev` branch, eventually. Marked as draft as long as tests are still failing.

### CODE REVIEW

**Comment by unbohn** [comment]
@pgbrodrick Some CLI tests are failing now due to a `tensorflow` related issue. The following error occurs when running `emulator = keras.models.load_model(self.engine_config.emulator_file)` in `sRTMnet.py`:

`ValueError: File format not supported: filepath=/home/runner/work/isofit/isofit/sRTMnet_v100/sRTMnet_v100. Keras 3 only supports V3 .keras files and legacy H5 format files (.h5 extension). Note that the legacy SavedModel format is not supported by load_model() in Keras 3. In order to reload a TensorFlow SavedModel as an inference-only layer in Keras 3, use keras.layers.TFSMLayer(/home/runner/work/isofit/isofit/sRTMnet_v100/sRTMnet_v100, call_endpoint='serving_default') (note that your call_endpoint might have a different name).`

Seems like the currently available emulator file is not compatible with keras version 3. What's your preference: should I implement the above-proposed fix or wait until #467 is merged?

**Comment by pgbrodrick** [comment]
Let's try and get #467 merged in.

**Comment by unbohn** [comment]
@pgbrodrick @jammont All CLI workflows are fixed! This PR is now ready for your review.

File: scripts/download-and-unpack-sRTMnet.sh
Comment (pgbrodrick) [inline scripts/download-and-unpack-sRTMnet.sh:56]: Good for now, to be fixed in #470.

**Comment by pgbrodrick** [comment]
Good for me.  Great work getting this complete @unbohn.  @jammont, when ready, please squash and merge.

File: isofit/utils/segment.py
Comment (jammont) [inline isofit/utils/segment.py:33]: With the addition of `ray.shutdown(); time.sleep()` in the test case, is this still necessary? Has it been tested if this impacts performance in any way?

File: .github/workflows/test-pythonpath-ubuntu.yml
Comment (jammont) [inline .github/workflows/test-pythonpath-ubuntu.yml:64]: Why `cp` instead of `mv`?

**Review by jammont — COMMENTED** [review]
Wonderful that the test cases are now passing again! Great job @unbohn, thanks for tackling this.

File: isofit/utils/segment.py
Comment (unbohn) [reply isofit/utils/segment.py:33]: @jammont I haven't tested if this impacts performance as the segmentation is generally pretty fast and not a bottleneck. You're probably right that this keyword is not needed as only the addition of `ray.shutdown(); time.sleep()` solved the issue. I'll remove it and see if the workflows still pass.

File: .github/workflows/test-pythonpath-ubuntu.yml
Comment (unbohn) [reply .github/workflows/test-pythonpath-ubuntu.yml:64]: For some reason, we need the `examples` folder both in the `isolation/` and the originating directory. I'd need to dig up the details, cannot say for sure why right now.

File: isofit/utils/segment.py
Comment (pgbrodrick) [reply isofit/utils/segment.py]: I think #472 might actually handle this, but ti doesn't hurt.  I've tested for performance, and this does not hurt us.

Comment (jammont) [reply isofit/utils/segment.py]: Discussed verbally at the team meeting: we'll be implementing this to all ray jobs in a future PR as we do not want any threading.

File: .github/workflows/test-pythonpath-ubuntu.yml
Comment (jammont) [reply .github/workflows/test-pythonpath-ubuntu.yml:64]: Discussed verbally offline: This will be fixed by integrating #469 into the test cases and workflows in a future PR.

**Review by jammont — APPROVED** [review]

**Comment by pl-kevinwurster** [comment]
Skimming through this PR and I notice there are three different ways of handling missing data files:

* One instance of Earth-sun-distance issues a [`logging.warning()`](https://github.com/isofit/isofit/pull/468/files#diff-21801ede7c46299deeb7284596c1290db904398357915bd1b65f9b22ddd9bd23R64), and populats an array with constant values. Note that another instances uses a `logging.Logger.info()`.
* Solar irradiance raises a [`FileNotFound()`](https://github.com/isofit/isofit/pull/468/files#diff-e68c4fecdaae5f543324083bacfe49befae8431b59b4f1155b1514ea40bbf50aR162).
* Another instance of Earth-sun-distance issues a [`logging.Logger.info()`](https://github.com/isofit/isofit/pull/468/files#diff-e68c4fecdaae5f543324083bacfe49befae8431b59b4f1155b1514ea40bbf50aR270), and populates an array with constant values. As noted, another instance issues a `logging.warning()`.

**Pull request merged**
